### PR TITLE
Fixed race condition causing deadlock.

### PR DESCRIPTION
**Pull request by emilwestergren**

### CODE REVIEW

**Comment by fredrikbryntesson** [comment]
Builds and run on both pc and android, memcheck ok, magic ok.

File: source/base/ThreadPool.ooc [hunk 1]
```diff
@@ -54,8 +55,8 @@ _ActionTask: class extends _Task {
 _ResultTask: class <T> extends _Task {
 	_result: Cell<T>
 	_action: Func -> T
-	init: func (=_action, =_mutex)
-	run: func (=_mutex) {
```
Comment (davidhesselbom) [inline source/base/ThreadPool.ooc]: Wow. Did I merge this? ...

**Comment by marcusnaslund** [comment]
My bad. 
There are other problems as well (like trying to join a thread more than once), which are still waiting in #509.

File: source/base/ThreadPool.ooc (hunk 1 above)
Comment (davidhesselbom) [inline source/base/ThreadPool.ooc]: Wait, I remember now. It wasn't as big a problem as I thought it was. Still, this new solution is much better.

**Pull request merged**